### PR TITLE
Cleaning the metamodel and the test.

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/optimization/optimizerlib.py [hunk 1]
```diff
@@ -1084,9 +1084,8 @@ def learn_on_k_best(archive: utils.Archive[utils.MultiValue], k: int) -> tp.Arra
 
     optimizer = Powell(parametrization=dimension, budget=45*dimension+30)
     try:
-        target = lambda x: float(model.predict(polynomial_features.fit_transform(np.asarray([x]))))
-        optimizer.minimize(target)
-        minimum = optimizer.provide_recommendation().value
+        minimum = optimizer.minimize(
+            lambda x: float(model.predict(polynomial_features.fit_transform(x.reshape((1, -1)))))).value
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:1088]:
```suggestion
            lambda x: float(model.predict(polynomial_features.fit_transform(x[None, :])))).value
```

File: nevergrad/optimization/test_optimizerlib.py [hunk 2]
```diff
@@ -438,22 +438,20 @@ def _ellips(x: np.ndarray) -> float:
...
 
     # Let us run the comparison.
-    default = (registry["CMA"] if dimension > 1 else registry["OnePlusOne"])(dimension, contextual_budget, num_workers=num_workers)
-    MetaModel = registry["MetaModel"](dimension, contextual_budget, num_workers=num_workers)
-    default_recom, MetaModel_recom = default.minimize(_target), MetaModel.minimize(_target) 
-    default_data = default_recom.get_standardized_data(reference=default.parametrization)
-    MetaModel_data = MetaModel_recom.get_standardized_data(reference=MetaModel.parametrization)
+    default = registry["CMA" if dimension > 1 else "OnePlusOne"](dimension, contextual_budget, num_workers=num_workers)
+    metamodel = registry["MetaModel"](dimension, contextual_budget, num_workers=num_workers)
+    default_recom, metamodel_recom = default.minimize(_target).value, metamodel.minimize(_target).value
```
Comment (jrapin) [inline nevergrad/optimization/test_optimizerlib.py:443]: Multiple long statements on a same line is very hard to read. Also, this is a lot of repeated code
```suggestion
recommendations: tp.List[np.ndarray] = []
for name in ("MetaModel", "CMA" if dimension > 1 else "OnePlusOne"):
    opt = registry[name](dimension, contextual_budget, num_workers=num_workers)
    recommendations.append(opt.minimize(_target).value)
metamodel_recom, default_recom = recommendations
```

File: nevergrad/optimization/test_optimizerlib.py [hunk 3]
```diff
@@ -438,22 +438,20 @@ def _ellips(x: np.ndarray) -> float:
...
 
     # With large budget, the difference should be significant.
     if budget > 60 * dimension:
-        assert _target(default_data) > 4 * _target(MetaModel_data)
+        assert _target(default_recom) > 4. * _target(metamodel_recom)
 
     # ... even more in the non ellipsoid case.
     if budget > 60 * dimension and not ellipsoid:
-        assert _target(default_data) > 7 * _target(MetaModel_data)
+        assert _target(default_recom) > 7. * _target(metamodel_recom)
```
Comment (jrapin) [inline nevergrad/optimization/test_optimizerlib.py:454]: even all this could probably be well factorized, but nevermind for now, it's readable

**Review by jrapin — CHANGES_REQUESTED** [review]

**Comment by jrapin** [comment]
Please fix the flakiness, we can't keep tests that randomly fail

**Comment by teytaud** [comment]
I've removed the ones that sometimes fail on the CI.
For each test, failure is rare, but with so many tests shit happens.

**Comment by teytaud** [comment]
We check here a property which is inherently stochastic. Checking by repeating 20 times and checking that it fails at most twice would be expensive. I remove some tests.

**Comment by teytaud** [comment]
I check with pytest --count=10 -n 30 -k etam nevergrad/optimization/test_optimizerlib.py

**Comment by teytaud** [comment]
20 replicas without crash. Sounds good to me.
I keep testing with more replicas.
Maybe for some reason tests in the CI are not equivalent to local tests.

File: nevergrad/optimization/test_optimizerlib.py [hunk 4]
```diff
@@ -407,20 +407,20 @@ def test_parallel_es() -> None:
 @pytest.mark.parametrize(
     "dimension, num_workers, scale, budget, ellipsoid",
     [
-    (2, 3, 1., 120, False),
+    (2, 8, 1., 120, False),
     #(2, 1, 8., 120, True),
-    (2, 3, 8., 70, False),
+    #(2, 3, 8., 70, False),
```
Comment (jrapin) [inline nevergrad/optimization/test_optimizerlib.py:412]: at some points the additional commented lines should just be removed

Comment (teytaud) [reply nevergrad/optimization/test_optimizerlib.py:416]: I don't remove them as I do love them but I add an explanation.

**Review by jrapin — APPROVED** [review]

**Comment by teytaud** [comment]
Also solving flakiness in bo_init.

**Pull request merged**